vendor/github.com/ethereum/go-ethereum/eth/downloader/downloader.go

```diff
@@ -476,12 +479,17 @@ func (d *Downloader) syncWithPeer(p *peerConnection, hash common.Hash, td *big.I
 	} else if d.mode == FullSync {
 		fetchers = append(fetchers, d.processFullSyncContent)
 	}
-	return d.spawnSync(fetchers)
+	return d.spawnSync(errCancelHeaderFetch, fetchers)
```

Comment:
what the point of having an error in method signature?

Reply:
@dshulyak Following existing pattern from `fetchParts`, since `spawnSync` is a generic helper method and shouldn't include knowledge about the specific error since that is the caller's knowledge.